# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowwatcher/nsWindowWatcher.cpp
# Commit: f42bb5b48c1b
# Full Hash: f42bb5b48c1b83323c2a8e76818de9ec902b33f0
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-07 03:53:09
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 4: Don't second-guess CHROME_{REMOTE,FISSION}_WINDOW flags, r=kmag
#   
#   This could cause issues, as we could incorrectly override the decision made by
#   nsWindowWatcher, and then override the decision again after the method returns.
#   This could cause the initial content browser to be configured incorrectly.
# ==============================================================================

diff -r 1ab9d22c73bb -r f42bb5b48c1b toolkit/components/windowwatcher/nsWindowWatcher.cpp
--- a/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Mon Apr 06 14:29:35 2020 +0000
+++ b/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Mon Apr 06 14:29:43 2020 +0000
@@ -527,6 +527,10 @@
 
   uint32_t chromeFlags = CalculateChromeFlagsForChild(aFeatures, sizeSpec);
 
+  if (isPrivateBrowsingWindow) {
+    chromeFlags |= nsIWebBrowserChrome::CHROME_PRIVATE_WINDOW;
+  }
+
   // A content process has asked for a new window, which implies
   // that the new window will need to be remote.
   chromeFlags |= nsIWebBrowserChrome::CHROME_REMOTE_WINDOW;
@@ -722,6 +726,10 @@
       if (docShell->UseRemoteSubframes()) {
         chromeFlags |= nsIWebBrowserChrome::CHROME_FISSION_WINDOW;
       }
+    } else if (XRE_IsContentProcess()) {
+      // If we don't have a parent window, but we're loaded in the content
+      // process, force ourselves into a remote window.
+      chromeFlags |= nsIWebBrowserChrome::CHROME_REMOTE_WINDOW;
     }
   }
 